sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v2/script.sql

```diff
@@ -0,0 +1,376 @@
+                  'firefox_mobile_download',
+                  'focus_download',
+                  'klar_download'
+                )
+                THEN 1
+              ELSE 0
+            END
+          ) AS boolean
+        ) AS had_download_event
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_2*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        e.key = 'ga_session_id'
+        AND e.value.int_value IS NOT NULL
+      GROUP BY
+        a.user_pseudo_id,
+        CAST(e.value.int_value AS string)
+    ),
+    stub_session_ids_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        event_timestamp,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        CAST(e.value.int_value AS string) AS stub_session_id
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_2*`
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        event_name = 'stub_session_set'
+        AND e.key = 'id'
+        AND e.value.int_value IS NOT NULL
+    ),
+    all_stub_session_ids AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        ARRAY_AGG(stub_session_id) AS all_reported_stub_session_ids,
+        ARRAY_AGG(stub_session_id ORDER BY event_timestamp DESC)[0] AS last_reported_stub_session_id
+      FROM
+        stub_session_ids_staging
+      GROUP BY
+        ga_client_id,
+        ga_session_id
+    ),
+    landing_page_by_session_staging AS (
+      SELECT
+        user_pseudo_id AS ga_client_id,
+        CAST(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'ga_session_id'
+            LIMIT
+              1
+          ).int_value AS string
+        ) AS ga_session_id,
+        SPLIT(
+          (
+            SELECT
+              `value`
+            FROM
+              UNNEST(event_params)
+            WHERE
+              key = 'page_location'
+            LIMIT
+              1
+          ).string_value,
+          '?'
+        )[OFFSET(0)] AS page_location,
+        event_timestamp
+      FROM
+        `moz-fx-data-marketing-prod.analytics_313696158.events_2*` a
+      JOIN
+        UNNEST(event_params) e
+      WHERE
+        e.key = 'entrances'
+        AND e.value.int_value = 1
+    ),
+    landing_page_by_session AS (
+      SELECT
+        ga_client_id,
+        ga_session_id,
+        page_location,
+        event_timestamp,
+        ROW_NUMBER() OVER (
+          PARTITION BY
+            ga_client_id,
+            ga_session_id
+          ORDER BY
+            event_timestamp ASC
+        ) AS lp_rnk
+      FROM
+        landing_page_by_session_staging
+      QUALIFY
+        lp_rnk = 1
```

Comment:
You can avoid creating the field in the select by having the now_number call inside the QUALIFY, here's an example:
https://github.com/mozilla/bigquery-etl/blob/e056243587055936570e0124d67686c4cc5ef1cd/sql/moz-fx-data-shared-prod/telemetry_derived/fog_decision_support_percentiles_v1/query.sql#L188-L189

Reply:
fixed now